diff --git a/node_modules/app-builder-lib/out/targets/nsis/NsisTarget.js b/node_modules/app-builder-lib/out/targets/nsis/NsisTarget.js
index e487f90..a09e531 100644
--- a/node_modules/app-builder-lib/out/targets/nsis/NsisTarget.js
+++ b/node_modules/app-builder-lib/out/targets/nsis/NsisTarget.js
@@ -473,7 +473,6 @@ class NsisTarget extends core_1.Target {
     }
     async executeMakensis(defines, commands, script) {
         const args = this.options.warningsAsErrors === false ? [] : ["-WX"];
-        args.push("-INPUTCHARSET", "UTF8");
         for (const name of Object.keys(defines)) {
             const value = defines[name];
             if (value == null) {
